cookbook/form/data_transformers.rst

```diff
@@ -4,29 +4,143 @@
-something else for use in your program. You could easily do this manually in your
-controller, but what if you want to use this specific form in different places?
-
-Say you have a one-to-one relation of Task to Issue, e.g. a Task optionally has an
-issue linked to it. Adding a listbox with all possible issues can eventually lead to
-a really long listbox in which it is impossible to find something. You might
-want to add a textbox instead, where the user can simply enter the issue number.
-
-You could try to do this in your controller, but it's not the best solution.
-It would be better if this issue were automatically converted to an Issue object.
-This is where Data Transformers come into play.
+Data transformers are used to translate the data for a field into a format that can
+be displayed in a form (and back on submit). They're already used internally for
+many field types. For example, the :doc:`date field type </reference/forms/types/date>`
+can be rendered as a ``yyyy-MM-dd``-formatted input textbox. Internally, a data transformer
+converts the starting ``DateTime`` value of the field into the ``yyyy-MM-dd`` string
+to render the form, and then back into a ``DateTime`` object on submit.
 
 .. caution::
 
     When a form field has the ``inherit_data`` option set, Data Transformers
     won't be applied to that field.
 
+Simple Example: Sanitizing HTML on User Input
+---------------------------------------------
+
+Suppose you have a Task form with a description ``textarea`` type::
+
+    // src/AppBundle/Form/TaskType.php
+    // ...
+
+    class TaskType extends AbstractType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('description', 'textarea');
+        }
+        
+        public function setDefaultOptions(OptionsResolverInterface $resolver)
+        {
+            $resolver->setDefaults(array(
+                'data_class' => 'AppBundle\Entity\Task'
+            ));
+        }
+
+        // ...
+    }
+
+But, there are two complications:
+
+#. Your users are allowed to use *some* HTML tags, but not others: you need a way
+   to call :phpfunction:`striptags` after the form is submitted.
+
+#. To be friendly, you want to convert ``<br/>`` tags into line breaks (``\n``) before
+   rendering the field so the text is easier to edit.
+
+This is a *perfect* time to attach a custom data transformer to the ``description``
+field. The easiest way to do this is with the :class:`Symfony\\Component\\Form\\CallbackTransformer`
+class::
+
+    // src/AppBundle/Form/TaskType.php
+
+    use Symfony\Component\Form\CallbackTransformer;
+    // ...
+
+    class TaskType extends AbstractType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('description', 'textarea');
+
+            $builder->get('description')
```

Comment:
Ha, indeed. This was on purpose. I think the one you propose (and the one we show now) is too hard and doesn't evolve smoothly: you usually start with `$builder->add('description' 'text')`. Then, if you need to add a transformer using the method you're showing, you have to re-write that line entirely (and I can never remember what this looks like myself).

Instead, why not add the field like normal, then fetch it to add the transformer?